packages/neon_framework/test/dialog_test.dart

```diff
@@ -0,0 +1,171 @@
+        expect(find.byType(OutlinedButton), findsOneWidget);
+        expect(find.byType(ElevatedButton), findsOneWidget);
+
+        dialog = const NeonConfirmationDialog(
+          title: title,
+          isDestructive: false,
+        );
+
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.byType(NeonDialogAction), findsExactly(2));
+        expect(find.byType(OutlinedButton), findsExactly(2));
+
+        const icon = Icon(Icons.error);
+        const content = SizedBox(key: Key('content'));
+        const confirmAction = SizedBox(key: Key('confirmAction'));
+        const declineAction = SizedBox(key: Key('declineAction'));
+        dialog = const NeonConfirmationDialog(
+          title: title,
+          icon: icon,
+          content: content,
+          confirmAction: confirmAction,
+          declineAction: declineAction,
+        );
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.byIcon(Icons.error), findsOneWidget);
+        expect(find.byKey(const Key('content')), findsOneWidget);
+        expect(find.byKey(const Key('confirmAction')), findsOneWidget);
+        expect(find.byKey(const Key('declineAction')), findsOneWidget);
+      });
+
+      testWidgets('NeonConfirmationDialog confirm', (final widgetTester) async {
+        const title = 'My Title';
+        const dialog = NeonConfirmationDialog(title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        await widgetTester.tap(find.byType(OutlinedButton));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+
+      testWidgets('NeonConfirmationDialog decline', (final widgetTester) async {
+        const title = 'My Title';
+        const dialog = NeonConfirmationDialog(title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        await widgetTester.tap(find.byType(ElevatedButton));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+    });
+
+    group('NeonRenameDialog', () {
+      testWidgets('NeonRenameDialog widget', (final widgetTester) async {
+        const title = 'My Title';
+        const value = 'My value';
+        const dialog = NeonRenameDialog(title: title, value: value);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.text(title), findsExactly(2), reason: 'The title is also used for the confirmation button');
+        expect(find.text(value), findsOneWidget);
+        expect(find.byType(TextFormField), findsOneWidget);
+      });
+
+      testWidgets('NeonRenameDialog actions', (final widgetTester) async {
+        const title = 'My Title';
+        const value = 'My value';
+        const dialog = NeonRenameDialog(title: title, value: value);
+
+        // Equal value should not submit
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+        await widgetTester.enterText(find.byType(TextFormField), '');
```

Comment:
Not needed to type an empty string

Reply:
ah wrong description. It's to explicitly check that "Empty values should not submit"